src/components/shared/Validations.tsx

```diff
@@ -81,48 +107,29 @@ const NumberValidations = ({
 const KeyValueValidations = ({ validations, className }: { validations: Dictionary<unknown>; className?: string }) => (
   <>
     {keys(validations)
-      .filter(validation => validation !== 'format')
+      .filter(key => Object.keys(validationFormatters).includes(key))
       .map(key => {
-        return <KeyValueValidation key={key} name={key} value={validations[key]} className={className} />;
+        const validation = validationFormatters[key](validations[key]);
+        if (validation) {
+          return (
+            <KeyValueValidation key={key} name={validation.name} values={validation.values} className={className} />
+          );
+        } else {
+          return null;
+        }
       })}
   </>
 );
 
-const KeyValueValidation = ({
-  className,
-  name,
-  value,
-}: {
-  className?: string;
-  name: string;
-  value: Dictionary<unknown> | unknown[] | unknown;
-}) => {
-  if (typeof value === 'object' && !Array.isArray(value) && value !== null) {
-    return (
-      <>
-        {keys(value).map(key => (
-          <KeyValueValidation key={key} className={className} name={`${name}.${key}`} value={value[key]} />
-        ))}
-      </>
-    );
-  }
-  const validation = Array.isArray(value) ? value : [value];
+const KeyValueValidation = ({ className, name, values }: { className?: string; name: string; values: string[] }) => {
   return (
     <Flex flexWrap color="muted" my={2} className={className}>
       <Text fontWeight="light">{capitalize(name)}:</Text>
-      {validation
-        .filter(
-          (v): v is Exclude<Primitive, null> | { value: string } =>
-            typeof v !== 'object' || (typeof v === 'object' && v !== null && 'value' in v),
-        )
-        .map(v => {
-          const value = typeof v === 'object' ? v.value : String(v);
-          return (
-            <Text key={value} ml={2} px={1} fontFamily="mono" border rounded="lg" className={className}>
-              {value}
-            </Text>
-          );
-        })}
+      {values.map(value => (
+        <Text key={value} ml={2} px={1} fontFamily="mono" border rounded="lg" className={className}>
+          {value}
+        </Text>
+      ))}
```

Comment:
Are you sure `value`s are unique? I know that's how things *generally* are, but I like to be very cautious with `key`s, because they cause hard-to-debug undefined behavior when non-unique. Please either run it through `uniq` or use another key, such as the index. Whichever you think fits the business requirements better.